h/static/scripts/tests/controllers/character-limit-controller-test.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+var CharacterLimitController = require('../../controllers/character-limit-controller');
+var util = require('./util');
+
+describe('CharacterLimitController', function () {
+
+  /**
+   * Make a <textarea> with the character limit controller enhancement
+   * applied and return the various parts of the component.
+   */
+  function component(value, maxlength) {
+    maxlength = maxlength || 250;
+
+    var template = '<textarea class="js-character-limit" data-maxlength="';
+    template += maxlength + '">';
+    if (value) {
+      template += value;
+    }
+    template += '</textarea><span class="js-character-counter">Foo</span>';
+
+    var containerEl = util.setupComponent(document, template, {
+      '.js-character-limit': CharacterLimitController,
+    });
+    var counterEl = containerEl.querySelector('.js-character-counter');
+    var textarea = containerEl.querySelector('.js-character-limit');
+    var ctrl = textarea.controllers[0];
+
+    return {
+      containerEl: containerEl,
+      counterEl: counterEl,
+      textarea: textarea,
+      ctrl: ctrl,
+    };
+  }
+
+  it('adds the ready class', function () {
+    var counterEl = component().counterEl;
+
+    assert.equal(counterEl.classList.contains('is-ready'), true);
```

Comment:
mocha provides a fairly minimal set of hooks in its core, on the assumption that you'll roll your own additional facilities as needed, eg:

```js
var afterFns;

beforeEach(function () {
  afterFns = [];
});

afterEach(function () {
  afterFns.forEach(function (fn) { fn(); });
});

it('creates a thing which needs to be cleaned up', function () {
  var thing = createThing();

  afterFns.push(function () { thing.remove(); });
});
```